src/core/components/HoverIntent/index.js

```diff
@@ -0,0 +1,76 @@
+import React, { PropTypes } from 'react';
+
+export default class HoverIntent extends React.Component {
+  static propTypes = {
+    sensitivity: PropTypes.number,
+    interval: PropTypes.number,
+    onHoverIntent: PropTypes.func.isRequired,
+    onHoverIntentEnd: PropTypes.func.isRequired,
+    children: PropTypes.element.isRequired,
+  }
+
+  static defaultProps = {
+    sensitivity: 5,
+    interval: 100,
+  }
+
+  componentWillUnmount() {
+    this.isHoverIntended = false;
+    this.clearHoverIntentDetection();
+  }
+
+  onMouseOver = (e) => {
+    e.persist();
+    const currentTarget = e.currentTarget;
+
+    const sq = (x) => x * x;
+    const distanceSq = (p1, p2) => sq(p1.x - p2.x) + sq(p1.y - p2.y);
+
+    const sensitivitySq = sq(this.props.sensitivity);
+
+    const initialPosition = { x: e.clientX, y: e.clientY };
+    let previousPosition = initialPosition;
+    this.currentMousePosition = initialPosition;
+
+    this.interval = setInterval(() => {
+      const currentPosition = this.currentMousePosition;
```

Comment:
If you move the closure into another function you should be able to test this synchronously.

```jsx
onMouseOver = (e) => {
  setInterval(this.createHoverIntentDetector(e), this.props.interval);
}

createHoverIntentDetector(e) {
  e.persist();
  const { currentTarget } = e;

  const sensitivitySq = sq(this.props.sensitivity);
  const initialPosition = { x: e.clientX, y: e.clientY };
  let previousPosition = initialPosition;
  this.currentMousePosition = initialPosition;

  return () => {
    if (distanceSq(initialPosition, currentPosition) > sensitivitySq &&
      distanceSq(previousPosition, currentPosition) < sensitivitySq) {
      this.clearHoverIntentDetection();
      this.isHoverIntended = true; 
      this.props.onHoverIntent(Object.assign({}, e, { currentTarget }));
    }
    previousPosition = currentPosition;
  }
}

it('triggers when the mouse slows down', () => {
  const e = { currentTarget, persist, x: 1, y: 2 };
  hoverIntentDetector = root.createHoverIntentDetector(e);
  Simulate.mouseMove({ x: 7, y: 7 }); // Moved too fast, not hover intent
  hoverIntentDetector();
  assert.isNotOk(props.hoverIntent.called);
  Simulate.mouseMove({ x: 8, y: 8 }); // Only moved a little, hover intent
  hoverIntentDetector();
  assert.ok(props.hoverIntent.called);
  
})